python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+        """
+
+        dust = sncosmo.OD94Dust()
+        sncosmo.Model.__init__(self, source=source, effects=[dust, dust],
+                               effect_names=['host', 'mw'],
+                               effect_frames=['rest', 'obs'])
+
+        # Current implementation of Model has a default value of mwebv = 0.
+        # ie. no extinction, but this is not part of the API, so should not
+        # depend on it, set explicitly in order to unextincted SED from
+        # SNCosmo. We will use catsim extinction from `lsst.sims.photUtils`.
+
+        self.ModelSource = source
+        self.set(mwebv=0.)
+
+        # ra and dec if passed are assumed to be in degrees and converted into
+        # radians.
+        self._ra = ra
+        self._dec = dec
+
+        # NB: More lines of code to support the possibility that ra, dec are
+        # not provided at instantiation, and default to None
+        self._hascoords = True
+        if self._dec is None:
+            self._hascoords = False
+        if self._ra is None:
+            self._hascoords = False
+
+        # Satisfied that coordinates provided are floats
+        if self._hascoords:
+            self.setCoords(ra, dec)
+
+        # For values of ra, dec, the E(B-V) is calculated directly
+        # from DustMaps
+        self.lsstmwebv = EBVbase()
+        self.ebvofMW = None
+        if self._hascoords:
+            self.mwEBVfromMaps()
+        return
+
+    @property
+    def SNstate(self):
+        """
+        Dictionary summarizing the property of the state. Can be used to
+        serialize to disk
+
+        Returns : Dictionary with values of parameters of the model.
+        """
+        statedict = dict()
+
+        # SNCosmo Parameters
+        statedict['ModelSource'] = self.source.name
+        for param_name in self.param_names:
+            statedict[param_name] = self.get(param_name)
+
+        # New Attributes
+        # statedict['lsstmwebv'] = self.lsstmwebv
+        statedict['_ra'] = self._ra
+        statedict['_dec'] = self._dec
+        statedict['MWE(B-V)'] = self.ebvofMW
+
+        return statedict
+
+    @staticmethod
+    def equivsncosmoParamDict(SNstate, SNCosmoModel):
+        """
+        return a dictionary that contains the parameters of SNCosmoModel
+        that are contained in SNstate
+
+        Parameters
+        ----------
+        SNstate : `SNObject.SNstate`, mandatory
+            Dictionary defining the state of a SNObject
+        SNCosmoModel : A `sncosmo.Model` instance, mandatory
+
+        Returns
+        -------
+        sncosmoParams: Dictionary of sncosmo parameters
+
+        """
+        sncosmoParams = dict()
+        for param in SNstate.keys():
+            if param in SNCosmoModel.param_names:
+                sncosmoParams[param] = SNstate[param]
+        sncosmoParams['mwebv'] = SNstate['MWE(B-V)']
+        return sncosmoParams
+
+    @staticmethod
+    def sncosmoParamDict(SNstate, SNCosmoModel):
```

Comment:
Why is it necessary that this be a separate method?  The only difference between this and equivsncosmoParamDict is that this method does not set sncosmoParams['mwebv'].  Is that important?